code/datums/components/vision_hurting.dm

```diff
@@ -0,0 +1,27 @@
+/// A component that damages eyes that look at the owner
+/datum/component/vision_hurting
+	var/damage_per_second
+	var/message
+
+/datum/component/vision_hurting/Initialize(damage_per_second=1, message="Your eyes burn as you look at")
+	if(!isatom(parent))
```

Comment:
why not `iscarbon()`? this code only works on that level

Reply:
brother the parent is the singularity

Reply:
Im stupid